docs/docs/reference/gen_notebooks/feedback_prod.md

```diff
@@ -68,60 +81,103 @@ def chat_response(prompt):
+                                    "👎"
+                                )
+                                st.success("Thanks for the feedback!")
+
+                    # Text feedback
+                    with col3:
+                        feedback_text = st.text_input(
+                            "Feedback", key=f"feedback_input_{idx}"
+                        )
+                        if st.button("Submit Feedback", key=f"submit_feedback_{idx}"):
+                            if feedback_text and call_idx < len(st.session_state.calls):
+                                # highlight-next-line
+                                st.session_state.calls[call_idx].feedback.add_note(
+                                    feedback_text
+                                )
+                                st.success("Feedback submitted!")
+
+
+def show_chat_prompt():
     if prompt := st.chat_input("What is up?"):
         st.session_state.messages.append({"role": "user", "content": prompt})
 
-        with st.chat_message("user"):
-            st.markdown(prompt)
-
         with st.chat_message("assistant"):
-            # highlight-next-line
             with weave.attributes(
                 {"session": st.session_state["session_id"], "env": "prod"}
             ):
-                # This could also be weave model.predict.call if you're using a weave.Model subclass
-                result, call = chat_response.call(
-                    prompt
-                )  # call the function with `.call`, this returns a tuple with a new Call object
-                # highlight-next-line
-                st.button(
-                    ":thumbsup:",
-                    on_click=lambda: call.feedback.add_reaction("👍"),
-                    key="up",
-                )
-                # highlight-next-line
-                st.button(
-                    ":thumbsdown:",
-                    on_click=lambda: call.feedback.add_reaction("👎"),
-                    key="down",
-                )
+                result, call = chat_response.call(prompt)
+                st.write(result["response"])
                 st.session_state.messages.append(
                     {"role": "assistant", "content": result["response"]}
                 )
+                # highlight-next-line
+                st.session_state.calls.append(call)
+
 
+def init_session_state():
+    if "session_id" not in st.session_state:
+        st.session_state["session_id"] = "123abc"
 
-def init_chat_history():
     if "messages" not in st.session_state:
-        st.session_state.messages = st.session_state.messages = []
+        st.session_state.messages = []
+
+    if "calls" not in st.session_state:
+        st.session_state.calls = []
 
 
 def main():
-    st.session_state["session_id"] = "123abc"
-    init_chat_history()
+    st.title("Add feedback")
+
+    init_session_state()
     display_chat_messages()
-    get_and_process_prompt()
+    show_chat_prompt()
+
+    st.rerun()
```

Comment:
what's the purpose of this addition? I remove this and it stops the constant refreshing